test/test_client.py

```diff
@@ -572,30 +588,39 @@ def test_as_read_area(self):
         dbnumber = 1
         data = bytearray(b'\x11')
         self.client.write_area(area, dbnumber, start, data)
-        res = self.client.as_read_area(area, dbnumber, start, amount)
-        self.client.wait_as_completion(10)
-        res2 = bytearray(res)
-        self.assertEqual(res2, data)
+        wordlen, usrdata = self.client._prepare_as_read_area(area, amount)
+        pusrdata = ctypes.byref(usrdata)
+        res = self.client.as_read_area(area, dbnumber, start, amount, wordlen, pusrdata)
+        if res != 0:
+            self.fail(f"Client failed: --  as_read_area command returned {res}")
```

Comment:
just do a `self.assertEqual(res, 0)`, no need to a custom error message

Reply:
also, doesn't make check_result() make sure it is always 0? so no      nee dfor the double check.

Reply:
I haven't looked at that since a while, but I guess this is a leftover while testing. I received sometimes values, which haven't been cought by check_result(). I guess the assert statement in error_text() needs to be replaced by `if context in ("client", "server", "partner")`, since assert can be ignored while running in process. And if there is something realy strange received, log that error.

But you are right, that this shouldn't be checked here again.